cmd/apiserver-watcher/azure_handler.go

```diff
@@ -0,0 +1,167 @@
+	"net"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/golang/glog"
+	"github.com/vishvananda/netlink"
+)
+
+// Prevent hairpin traffic when the apiserver is up
+
+// As per the Azure documentation (https://docs.microsoft.com/en-us/azure/load-balancer/concepts//limitations),
+// if a backend is load-balanced to itself, then the traffic will be dropped.
+//
+// This is because the L3LB does DNAT, so while the outgoing packet has a destination
+// IP of the VIP, the incoming load-balanced packet has a destination IP of the
+// host. That means that it "sees" a syn with the source and destination
+// IPs of itself, and duly replies wit a syn-ack back to itself. However, the client
+// socket expects a syn-ack with a source IP of the VIP, so it drops the packet.
+//
+// The solution is to redirect traffic destined to the lb vip back to ourselves.
+//
+// Example rules using vip 10.0.0.1:
+//
+// -A PREROUTING -m comment --comment "azure LB vip overriding for pods" -j azure-vips
+// -A OUTPUT -m comment --comment "azure LB vip overriding for local clients" -j azure-vips
+// -A azure-vips --dst 10.0.0.1 -j REDIRECT
+
+type azureHandler struct {
+	vip    string
+	ipt    *iptables.IPTables
+	stopCh chan struct{}
+}
+
+var _ handler = &azureHandler{}
+
+func newAzureHandler(vip string, isIPv6 bool) (handler, error) {
+	proto := iptables.ProtocolIPv4
+	if isIPv6 {
+		proto = iptables.ProtocolIPv6
+	}
+
+	ipt, err := iptables.New(
+		iptables.IPFamily(proto),
+		iptables.Timeout(iptablesTimeout),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &azureHandler{
+		vip:    vip,
+		ipt:    ipt,
+		stopCh: make(chan struct{}),
+	}, nil
+}
+
+// onFailure: either stop the routes service, or write downfile
+func (az *azureHandler) onFailure() error {
+	select {
+	case az.stopCh <- struct{}{}:
+		glog.V(4).Info("stopping iptables sync loop")
+	default:
+		glog.V(4).Info("no sync iptables running")
+	}
+	// wait for iptables sync loop to finish
+	time.Sleep(iptablesTimeout * time.Second)
+	return az.deleteRules()
+}
+
+// onSuccess: install iptables rules to REDIRECT LB traffic to the apiserver
+func (az *azureHandler) onSuccess() error {
+	if err := az.syncRulesOnce(); err != nil {
+		return err
+	}
+	// bz1930457 delete stale conntrack entries to the apiserver
+	ip := net.ParseIP(az.vip)
+	var family netlink.InetFamily
+	family = netlink.FAMILY_V4
+	if az.ipt.Proto() == iptables.ProtocolIPv6 {
+		family = netlink.FAMILY_V6
+	}
+	filter := &netlink.ConntrackFilter{}
+	filter.AddIP(netlink.ConntrackNatSrcIP, ip)
+	if _, err := netlink.ConntrackDeleteFilter(netlink.ConntrackTable, family, filter); err != nil {
+		glog.V(4).Infof("Error deleting conntrack entries for %s", ip.String())
+	}
+
+	go az.syncRulesUntil()
+	return nil
+
+}
+
+// https://github.com/openshift/installer/blob/master/data/data/azure/vnet/internal-lb.tf
+// interval_in_seconds = 5
+// number_of_probes    = 2
+func (az *azureHandler) successThreshold() int {
+	return 1
```

Comment:
why 1 ?

Reply:
those were the previous values, https://github.com/openshift/machine-config-operator/pull/2422/commits/bcba00705b625c0fb230b0a5cb8edeb880b0107d#r584729261
I kept them